### PR TITLE
Add adaptive mask plot to report

### LINKED ISSUE #1072: Add adaptive mask figure to the HTML report
### Summary

Here's a draft using [`nilearn.plotting.plot_prob_atlas()`](https://nilearn.github.io/stable/modules/generated/nilearn.plotting.plot_prob_atlas.html#nilearn.plotting.plot_prob_atlas).

adaptive_mask

### PR DESCRIPTION
**Pull request by tsalo**
Closes #1072.

Changes proposed in this pull request:

- Create a new function named `tedana.reporting.static_figures.plot_adaptive_mask`, which creates a figure showing the base mask, the denoising mask (adaptive mask >= 1), and the classification mask (adaptive mask >= 3) overlaid on top of the mean optimally combined image.
- Add the adaptive mask figure to the HTML report.

### CODE REVIEW

**Review by eurunuela — CHANGES_REQUESTED** [review]
Changes look good to me. The only thing I would change is I would make the figure bigger with some css like `style='height: 500px'` or something like that.

**Comment by eurunuela** [comment]
I think that's better. I noticed the figure has some blank area above and below the actual figure itself. This means that effectively, the brains are not going to be 500px in height, but the whole thing with the blank area above and below will be. I don't know where this blank area came from. It must be from the function that generates the image.

**Comment by tsalo** [comment]
I dropped the `brainplot` class and things look better. @eurunuela WDYT?

**Comment by tsalo** [comment]
I'm working on improving the figure- the 3-echo one was looking weird.

**Comment by tsalo** [comment]
Okay it's looking a lot better now.

Screenshot 2024-04-12 at 10 55 37 AM

**Review by handwerkerd — CHANGES_REQUESTED** [review]
The code all looks fine. This will be a useful addition.

Just add explanatory documentation and a picture to the docs. The meaning of the three contours for `base` `optimal combination` and `classification` might not be obvious so explain a bit what they mean.
https://github.com/tsalo/tedana/blob/ab3cbdfbdcd4bac3e4dd42788500c8ad50e0e5a3/docs/outputs.rst#L309

**Comment by tsalo** [comment]
@handwerkerd WDYT? https://tedana--1073.org.readthedocs.build/en/1073/outputs.html#adaptive-mask-summary-plot

File: docs/outputs.rst [hunk 1]
```diff
@@ -528,11 +528,30 @@ should not overly focus on carpet plots and should examine these results in cont
...
+
+Below the carpet plots is a summary plot of the adaptive mask.
+
+This figure overlays contours reflecting the boundaries of the following masks onto the mean optimally combined data:
+
+- **Base**: The base mask, either provided by the user or generated automatically using ``compute_epi_mask``.
+- **Optimal combination**: The mask used for optimal combination and denoising.
+  This corresponds to values greater than or equal to 1 in the adaptive mask.
+- **Classification**: The mask used for the decomposition and component classification steps.
+  This corresponds to values greather than or equal to 3 in the adaptive mask.
```
Comment (handwerkerd) [inline docs/outputs.rst:543]:
```suggestion
  This corresponds to values greater than or equal to 1 (at least 1 good echo) in the adaptive mask.
- **Classification**: The mask used for the decomposition and component classification steps.
  This corresponds to values greather than or equal to 3 (at least 3 good echoes) in the adaptive mask.
```

**Review by handwerkerd — COMMENTED** [review]
Minor, optional change of wording. Just wanted to clarify what 1 & 3 mean.

**Review by handwerkerd — APPROVED** [review]
LGTM!

**Review by eurunuela — APPROVED** [review]
This is great. Thank you @tsalo!

**Pull request merged**